plugins/services/src/js/structs/ServiceTree.js

```diff
@@ -370,4 +370,13 @@ module.exports = class ServiceTree extends Tree {
       return serviceTreeLabels;
     }, []);
   }
+
+  getGroupRolesLength() {
```

Comment:
I think we should have 1 function that returns the could of both group and non-group roles so that we only have to walk the tree a single time to get those counts.

Reply:
We should also add unit tests for that function and make sure we test for deeply nested groups, apps, services and Pods.

Reply:
We can do both of these things.